### ISSUE TITLE
App darkening on multi monitor setup

### ISSUE THREAD

**Issue by MeneerDoktoor** [PapyElGringo/material-shell#117]
I use the external monitor for movies etz, usualy stuff i'm not working on but want visible at all times. I can tell you darken the apps that are not being used.
![alt text](https://i.ibb.co/vJYV8GX/Screenshot-from-2019-09-29-23-39-51.png)
Would be lovely to have an option to disable it the feature on a or all monitor(s).

Can't seem to find the function sifting trough the code though. For the time being, could you point me in the right direction?

Amazing work so far though, cheers!

**Comment by codycraven** [comment]
I encounter the same thing when in a call on Slack (on a single monitor) and it drives me bonkers---basically darkens out all windows that are not Slack when a participant is screen sharing. Now that I see this is related to material-shell, I'll try to dig in and change it.

**Comment by codycraven** [comment]
Update, I grepped the code for this shade effect and nothing stood out, I searched through the different CSS rgba rules, looked for the words "shade", "dim", and "darken" and came up dry. However I think tried "light" (as in highlight) and found widget/backdrop.js. My guess it is this file that's causing the dimming of inactive windows/monitors (I notice it when I have a floating window from a screenshare Google Meet or Slack).

I'll play with trying to disable this functionality and see what happens.

**Comment by codycraven** [comment]
@MeneerDoktoor this ended up working for me, can you try it for your issue and see if it resolves the problem?

Apply this change to your repo:
```diff
diff --git a/stylesheet.css b/stylesheet.css
index 97251ce..91f52e0 100644
--- a/stylesheet.css
+++ b/stylesheet.css
@@ -121,4 +121,4 @@
   margin: 8px; }
 
 .backdrop-container {
-  background: rgba(0, 0, 0, 0.5); }
+  background: rgba(0, 0, 0, 0.0); }
```

Restart X11 or log out and log back in (if on Wayland or X11). If that fixes it I think we have a good candidate to create an option for material-shell --- note my solution isn't an elegant one and is just to see if this resolves both of our issues.

**Comment by nununoisy** [comment]
If I'm reading correctly, this just hides the overlay with CSS, which is a hacky solution at best.
I might be able to quickly write a patch to account for this change in JS.

Although the real bug lies in that the overlay created for a dialog box is not contained to one workspace on the grid, but the entire grid. I unfortunately don't have a second monitor to test on so I don't think I can fix it.

**Comment by codycraven** [comment]
@nununoisy that's why I said this wasn't an elegant solution, it's just to see if it resolves the issue so that a better solution may be developed.

From what I understand (and what solved my problem) the solution would be to:

- Add a new Toggle in Material Shell's "Tiling settings" to allow the user to control whether the transparent backdrop is used or not.
- Make a change in tilingManager/tilingLayouts/baseTiling.js that omits the backdrop when the backdrop is disabled.

However this may not be the right solution, I'm hoping @MeneerDoktoor can confirm this fixes his problem before the effort is put in.

**Comment by codycraven** [comment]
@nununoisy re-reading with your extra information about the workspace and grid isolation I believe my observed problem and solution (while I think it would solve @MeneerDoktoor's core problem) is separate from that work that would fix the backdrop isolation issue you described.

I'd be happy to test your solution (since I'm not well enough aware of the codebase to solve it). I can use my laptop monitor to test (though I do not use it on a regular basis).

**Comment by MeneerDoktoor** [comment]
Yeah it's a good band-aid fix for now. Thanks @codycraven.

**Comment by nununoisy** [comment]
@MeneerDoktoor Try out my fix (#137) and see if this works for you.

**Comment by codycraven** [comment]
@MeneerDoktoor the configuration change is now available in latest master, #137 provides the toggle.
